### DISCUSSION TITLE
Question about mapping "repeated string" to "List<>" in example

### DISCUSSION THREAD

**Discussion by YoungHoney** [mapstruct/mapstruct#3658] [Q&A] [answered]
I have a question about gRPC and Mapstruct that I'd like to ask. I'm interested in using Mapstruct with gRPC, and I found an example here:

[https://github.com/mapstruct/mapstruct-examples/blob/main/mapstruct-on-gradle-protobuf3/usage/src/main/java/org/mapstruct/example/mapper/UserMapper.java](https://github.com/mapstruct/mapstruct/discussions/url)

In this example, UserDTO (in .proto) has "repeated PermissionDTO permissions", and User.java has "List<Permission> permissions".

but in UserMapper.java, 

  ` @Mapping(source = "permissions", target = "permissionsList") `
...
 `  UserDTO map(User user);`

   
I'm curious why the target parameter is permissions**List**. Why is "List" added to the name? If you have any links or tutorials explaining why "List" is added, I would greatly appreciate it.

**Comment by filiphr** [comment, marked as answer]
>I'm curious why the target parameter is permissionsList. Why is "List" added to the name? If you have any links or tutorials explaining why "List" is added, I would greatly appreciate it.

`List` is added to the name because the protobuf java plugin generates a getter named `getPermissionsList`, which means that the property from the side of MapStruct is called `permissionsList`. See the [Protobuf Java Docs](https://protobuf.dev/getting-started/javatutorial/#protobuf-api).

If you want to use protobuf, you might be interested in looking into the [mapstrict-spi-protobuf](https://github.com/entur/mapstruct-spi-protobuf). This is a project form the community and it is not maintained by the MapStruct team.

**Reply by YoungHoney** [reply]
thank you !
